ext/standard/string.c

```diff
@@ -3752,6 +3752,141 @@ PHPAPI zend_string *php_str_to_str(char *haystack, size_t length, char *needle,
 }
 /* }}} */
 
+/* {{{ php_str_to_array_ex
+ */
+PHPAPI zend_string *php_str_to_array_ex(char *haystack, size_t length,
+	char *needle, size_t needle_len, HashTable *arr, int case_sensitivity, size_t *replace_count)
+{
+	zval *entry;
+	zend_string *new_str;
+	HashPosition pos;
+	size_t new_length, offset, str_len;
+	char *checked_haystack, *checked_needle, *haystack_dup = NULL
+		, *needle_dup = NULL, *end, *p, *r, *n, *str;
+	size_t max_len=0, count=0;
+
+	/* Convert replace elements to string before entering the replace loop */
+	/* and compute max length */
+
+	ZEND_HASH_FOREACH_VAL(arr, entry) {
+		SEPARATE_ZVAL(entry);
+		convert_to_string(entry);
+		if (Z_STRLEN_P(entry) > max_len) max_len = Z_STRLEN_P(entry);
+	} ZEND_HASH_FOREACH_END();
+	zend_hash_internal_pointer_reset_ex(arr, &pos);
```

Comment:
Sorry, I don't understand what you mean. A first pass is done to convert array elements to string, then internal pointer is reset. What's wrong here ?